src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlDecimalTypeReference.cs

```diff
@@ -11,11 +11,29 @@ namespace Microsoft.OData.Edm.Csdl.Parsing.Ast
     /// </summary>
     internal class CsdlDecimalTypeReference : CsdlPrimitiveTypeReference
     {
-        public CsdlDecimalTypeReference(int? precision, int? scale, string typeName, bool isNullable, CsdlLocation location)
+        public CsdlDecimalTypeReference(int? precision, int? scale, string typeName, bool isNullable, CsdlLocation location, bool shouldWriteDefaultScale = false)
             : base(EdmPrimitiveTypeKind.Decimal, typeName, isNullable, location)
         {
             this.Precision = precision;
             this.Scale = scale;
+            this.ShouldWriteDefaultScale = shouldWriteDefaultScale;
         }
+
+        /// <summary>
+        /// Specifies whether the default scale should be written out explicitly
+        /// when the model is being serialized into a CSDL (XML) document.
+        /// </summary>
+        /// <remarks>
+        /// Usually the attribute is not written when it contains the default value.
+        /// This flag is only here for backwards compatibility:
+        /// Up to v7.10.0, the default scale was 0. This is consistent with the XML CSDL spec (but not JSON spec),

```

Comment:
`(but not JSON spec)` - Not sure I understand how writing a default scale of 0 was inconsistent with JSON specification...

Reply:
In CSDL JSON the default scale is null, not 0.

Reply:
argh, so you meant CSDL JSON, not JSON...